tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/Metrics.java

```diff
@@ -49,10 +50,62 @@
    */
   public static <T extends TNumber> Operand<T> topKCategoricalAccuracy(
       Ops tf, Operand<? extends TNumber> labels, Operand<T> predictions, long k) {
-    Operand<TFloat32> fPredictions = CastHelper.cast(tf, predictions, TFloat32.class);
-    return CastHelper.cast(
+    Operand<TFloat32> fPredictions = cast(tf, predictions, TFloat32.class);
+    return cast(
         tf,
         tf.nn.inTopK(fPredictions, tf.math.argMax(labels, tf.constant(-1)), tf.constant(k)),
         predictions.type());
   }
+
+  /**
+   * Computes how often integer targets are in the top K predictions.
+   *
+   * <p>Standalone usage:
+   *
+   * <pre>
+   *     Operand&lt;TInt32&gt; labels = tf.constant(new int[]{2, 1});
+   *     Operand&lt;TFloat32&gt; predictions = tf.constant(new float[][]
+   *                            {{0.1f, 0.9f, 0.f8}, {0.05f, 0.95f, 0f}});
+   *     Operand&lt;TFloat32&gt; m = Metrics.topKCategoricalAccuracy(
+   *                                    labels, predictions, 3)
+   *     //m.shape().toString == "[2]"
+   * </pre>
+   *
+   * @param tf the TensorFlow Ops.
+   * @param labels the ground truth values.
```

Comment:
Document the expected values and dimensions?
* `labels` is expected to be (batch_size), where each value is a class id. Any additional dimensions of size 1 will be squeezed.
* `predictions` is expected to be (batch_size, ...), where dimensions beyond the first represent class logits or probabilities.

In truth, we don't require `labels` and `predictions` to share a `batch_size` dimension; we only require them to be reshapable to that. I could imagine explicitly checking that they share a `batch_size` dimension, or documenting the true, looser requirement, or just letting that part go.